tools/matRad_compareDose.m

```diff
@@ -0,0 +1,166 @@
+%                   evaluated for every structure listed in 'cst'.
+%                   note that only voxels exceeding the dose threshold are
+%                   considered.
+%   hfig:           Figure handle struct for all 3 figures and subplots,
+%                   indexed by plane names
+%
+% References gamma analysis:
+%   [1]  http://www.ncbi.nlm.nih.gov/pubmed/9608475
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+% Copyright 2015 the matRad development team. 
+% 
+% This file is part of the matRad project. It is subject to the license 
+% terms in the LICENSE file found in the top-level directory of this 
+% distribution and at https://github.com/e0404/matRad/LICENSES.txt. No part 
+% of the matRad project, including this file, may be copied, modified, 
+% propagated, or distributed except according to the terms contained in the 
+% LICENSE file.
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%% check if cubes consistent
+if ~isequal(size(cube1),size(cube2))
+   error('dose cubes must be the same size\n'); 
+end
+
+if ~exist('localglobal','var')
+    localglobal = 'global';
+end
+if ~exist('n','var')
+    n = 0;
+end
+if ~exist('criteria','var')
+    criteria = [3 3];
+end
+%%
+resolution = [ct.resolution.x ct.resolution.y ct.resolution.z];
+
+%% Calculate iso-center slices
+isoCenter=matRad_getIsoCenter(cst,ct,0);
+slicename ={round(isoCenter(2)./resolution(2)),round(isoCenter(1)./resolution(1)),round(isoCenter(3)./resolution(3))};
+doseWindowCube2 = [0 max([cube2(:); cube2(:)])];
+doseWindowCube1 = [0 max([cube1(:); cube1(:)])];
+planename={'coronal','sagittal','axial'};
+
+%% Get the gamma cube
+disp('Calculating gamma index cube...');
+if exist('criteria','var')
+    relDoseThreshold = criteria(1); % in [%]
+    dist2AgreeMm     = criteria(2); % in [mm]
+else
+    dist2AgreeMm     = 3; % in [mm]
+    relDoseThreshold = 3; % in [%]
+end
+
+[gammaCube,gammaPassRate] = matRad_gammaIndex(cube1,cube2,resolution,criteria,[],n,localglobal,cst);
+set(gcf,'visible','off')
+
+
+%%% Calculate absolute difference cube and dose windows for plots
+DifferenceCube=abs(cube2-cube1);
+doseWindowCube3 = [0 max([DifferenceCube(:); DifferenceCube(:)])];
+gammadoseWindowCube = [0 max([gammaCube(:); gammaCube(:)])];
+
+%% Plot everything
+% Plot dose slices
+for plane=1:3;
+disp(['Plotting ',planename{plane},' plane...']);
+
+% Initialize Figure
+hfig.(planename{plane}).('fig')=figure('Renderer', 'painters', 'Position', [10 50 800 800]);
+set(gcf,'Color',[1 1 1]);
+
+% Plot Dose 1
+hfig.(planename{plane}).('cube1').Axes=subplot(2,2,1);
+    [hfig.(planename{plane}).('cube1').CMap,hfig.(planename{plane}).('cube1').Dose,hfig.(planename{plane}).('cube1').Ct,hfig.(planename{plane}).('cube1').Contour,hfig.(planename{plane}).('cube1').IsoDose]=matRad_plotSliceWrapper(gca,ct,cst,1,cube1,plane,slicename{plane},[],[],colorcube,[],doseWindowCube1,[],100);
```

Comment:
would be cool if this call would be felxible to work also with empty ct and/or cst structs...